plugins/modules/vmware_host_graphics.py

```diff
@@ -0,0 +1,190 @@
+---
+module: vmware_host_graphics
+version_added: '3.10.0'
+short_description: Manage Host Graphic Settings
+description:
+- This module can be used to manage Host Graphic Settings
+author:
+- Alexander Nikitin (@ihumster)
+options:
+  cluster_name:
+    description:
+    - Name of cluster.
+    - All host system from given cluster used to manage Host Graphic Settings.
+    - Required parameter, if C(esxi_hostname) is not set.
+    type: str
+  esxi_hostname:
+    description:
+    - List of ESXi hostname to manage Host Graphic Settings.
+    - Required parameter, if C(cluster_name) is not set.
+    type: list
+    elements: str
+  graphic_type:
+    description:
+    - Default graphics type
+    default: shared
+    choices: [ shared, sharedDirect ]
+    type: str
+  assigment_policy:
+    description:
+    - Shared passthrough GPU assignment policy
+    default: performance
+    choices: [ consolidation, performance ]
+    type: str
+  restart_xorg:
+    description:
+    - Restart X.Org Server after change any parameter ( C(graphic_type) or C(assigment_policy) )
+    default: False
+    type: bool
+extends_documentation_fragment:
+- community.vmware.vmware.documentation
+
+'''
+
+EXAMPLES = r'''
+- name: Change Host Graphics Settings
+  community.vmware.vmware_host_graphics:
+    hostname: '{{ vcenter_hostname }}'
+    username: '{{ vcenter_username }}'
+    password: '{{ vcenter_password }}'
+    esxi_hostname: '{{ esxi_hostname }}'
+    graphic_type: sharedDirect
+    assigment_policy: consolidation
+  delegate_to: localhost
+'''
+
+RETURN = r'''
+results:
+  description:
+    - data about host system graphics settings.
+  returned: always
+  type: dict
+  sample: {
+      "changed": true,
+      "esxi01": {
+          "changed": false,
+          "msg": "All Host Graphics Settings already configured"
+      },
+      "esxi02": {
+          "changed": true,
+          "msg": "New host graphics settings changed to: hostDefaultGraphicsType = 'shared', sharedPassthruAssignmentPolicy = 'performance'.X.Org was restarted"
+      }
+  }
+'''
+
+try:
+    from pyVmomi import vim
+    HAS_PYVMOMI = True
+except ImportError:
+    HAS_PYVMOMI = False
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible_collections.community.vmware.plugins.module_utils.vmware import vmware_argument_spec, PyVmomi
+from ansible.module_utils._text import to_native
+
+
+class VMwareHostGraphicSettings(PyVmomi):
+    """ Main class for configuring Host Graphics Settings """
+    def __init__(self, module):
+        super(VMwareHostGraphicSettings, self).__init__(module)
+        self.graphic_type = self.params.get('graphic_type', 'shared')
+        self.assigment_policy = self.params.get('assigment_policy', 'performance')
+        self.restart_xorg = self.params.get('restart_xorg', True)
+        self.results = {"changed": False}
+        esxi_hostname = self.params.get('esxi_hostname', None)
+        cluster_name = self.params.get('cluster_name', None)
```

Comment:
```suggestion
        self.graphic_type = module.params['graphic_type']
        self.assigment_policy = module.params['assigment_policy']
        self.restart_xorg = module.params['restart_xorg']
        self.results = {"changed": False}
        esxi_hostname = module.params['esxi_hostname']
        cluster_name = module.params['cluster_name']
```
I think it should be possible to access `module.params` here directly. Additionally, you shouldn't need to specify any default values here since ansible uses the defaults defined in the module spec. So, for example, graphic_type should be shared if unset, anyway.